### PR TITLE
Add a build flag for gpcloud

### PR DESCRIPTION
**Pull request by adam8157**
This commit does things below to make developers' life, who are still
using old version C++ compilers, easier.

Add a flag to optionally disable gpcloud compilation.
Add a configure check for C++11 if gpcloud is enabled.

Signed-off-by: Adam Lee <ali@pivotal.io>
Signed-off-by: Haozhou Wang <hawang@pivotal.io>

### CODE REVIEW

**Comment by schubert** [comment]
Seems okay at a glance. @pivotal-mike do you have any more thoughts on this? You've been asking for build flags like this.

File: GNUmakefile.in [hunk 1]
```diff
@@ -29,6 +29,7 @@ all:
 	$(MAKE) -C contrib/pg_upgrade all
 	$(MAKE) -C contrib/pgcrypto all
 	$(MAKE) -C gpAux/extensions mapreduce
+	$(MAKE) -C gpAux/extensions mkgpcloud
```
Comment (danielgustafsson) [inline GNUmakefile.in]: While not the fault of this PR, we should probably move to just doing `$(MAKE) -C gpAux/extensions all` and have that Makefile figure out what to do. Since we need the same for `contrib`, we can do that in a separate PR though.
